pyomo/future.py

```diff
@@ -28,14 +26,49 @@
 
 """
 
+from typing import Any, Literal, overload
+
+import pyomo.contrib.appsi.base as _appsi
+import pyomo.contrib.solver.common.factory as _contrib
+import pyomo.environ as _environ
+import pyomo.opt.base.solvers as _solvers
+
+_SolverFactoryClassV1 = _solvers.SolverFactoryClass
+_SolverFactoryClassV2 = _appsi.SolverFactoryClass
+_SolverFactoryClassV3 = _contrib.SolverFactoryClass
+
+solver_factory_v1: _SolverFactoryClassV1 = _solvers.LegacySolverFactory
+solver_factory_v2: _SolverFactoryClassV2 = _appsi.SolverFactory
+solver_factory_v3: _SolverFactoryClassV3 = _contrib.SolverFactory
+
+_versions = {1: solver_factory_v1, 2: solver_factory_v2, 3: solver_factory_v3}
```

Comment:
Future is intended to manage all "future compatibility changes", so we should use a more specific name than `versions` here - there will eventually be several "versions" that this module will manage.

I would prefer *not* promoting this dict to the module scope: `solver_factory` will probably never be called more than once, so the performance gain of preserving the dict isn't worth the memory.